dialoguekit/platforms/platform.py

```diff
@@ -1,25 +1,47 @@
 """The Platform facilitates displaying of the conversation."""
 
+from abc import ABC, abstractmethod
+from typing import Callable
+
 from dialoguekit.core.utterance import Utterance
 
 
-class Platform:
+class Platform(ABC):
     def __init__(self):
         """Represents a platform."""
-        pass
+        self._user_callback: Callable[[str], None] = None
+
+    def register_user_callback(self, callback: Callable[[str], None]) -> None:
+        """Registers a callback function to be called on user input.
+
+        Args:
+            callback: Function to call on user input
+        """
+        self._user_callback = callback
+
+    @abstractmethod
+    def listen_for_user_input(self) -> None:
+        """Listens for the user input.
```

Comment:
Nit: Maybe a Raises note can be added to doctrings.

Reply:
Done.